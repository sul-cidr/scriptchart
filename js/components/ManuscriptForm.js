import React from "react";

// In this component, we would actually pass a list of
// manuscripts, and have a toggle to sort by date or name
// The list could come from a json object with name and date
const ManuscriptForm = () => {
  return (
    <form className={"manuscript-form"}>
      <div className={"field"}>
        <label className={"control"}>Select Manuscripts:</label>
        <div className={"control"}>
          <div className={"select is-multiple"}>
            <select multiple>
              <option>Manuscript A</option>
              <option>Manuscript B</option>
              <option>Manuscript C</option>
              <option>Manuscript D</option>
              <option>Manuscript E</option>
              <option>Manuscript F</option>
            </select>
          </div>
        </div>
        <div className={"control sort-option"}>
          <p>Order manuscripts by:</p>
          <label className={"radio"}>
            Shelfmark:
            <input type={"radio"} name={"Shelfmark"} defaultChecked />
          </label>
          <label className={"radio"}>
            Date:
            <input type={"radio"} name={"Date"} />
          </label>
        </div>
      </div>
      <div className={"field"}>
        <label className={"control"}>Select Letters:</label>
        <div className={"control"}>
          <div className={"select is-multiple"}>
            <select multiple>
              <option>ʾĀlep̄</option>
              <option>Bēṯ</option>
              <option>Gāmal</option>
              <option>Dālaṯ</option>
              <option>Hē</option>
              <option>Waw</option>
            </select>
          </div>
        </div>
      </div>
      <div className={"field"}>
        <label className={"control"}>Select number of letter examples:</label>
        <div className={"control"}>
          <div className={"select"}>
            <select>
              <option>1</option>
              <option>2</option>
              <option>3</option>
            </select>
          </div>
        </div>
      </div>
      <div className={"field"}>
        <label className={"checkbox"}>
          Show binarized images?
          <input type={"checkbox"} />
        </label>
      </div>
      <div className={"field"}>
        <label className={"control"}>Select image size:</label>
        <div className={"control"}>
          <div className={"select"}>
            <select>
              <option>Small</option>
              <option>Medium</option>
              <option>Large</option>
            </select>
          </div>
        </div>
      </div>
      <div className={"field"}>
        <div className={"control"}>
          <button className={"button is-link"}>Submit</button>
        </div>
      </div>
    </form>
  );
};

export default ManuscriptForm;
