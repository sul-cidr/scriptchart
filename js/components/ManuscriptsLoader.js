import React from "react";

/* Mock data to use in development when local API backend is not available */
const fallbackManuscripts = [
  { id: 20, shelfmark: "BL. Add. 12146", date: "1007 " },
  { id: 21, shelfmark: "BL. Or. 8732", date: "770" },
  { id: 22, shelfmark: "BL. Or. 8606", date: "723" },
  { id: 23, shelfmark: "BL. Add. 17102", date: "598-599" },
  { id: 24, shelfmark: "BL. Add. 17107", date: "540-541" },
  { id: 25, shelfmark: "BL. Add. 12175", date: "533-534 " },
  { id: 26, shelfmark: "BL. Add. 17109", date: "873 - 874" }
];

class ManuscriptsLoader extends React.Component {
  constructor(props) {
    super(props);
    this.state = { manuscripts: [] };
  }
  componentDidMount() {
    fetch("http://localhost:8000/api/manuscripts?format=json")
      .then(response => {
        return response.json();
      })
      /* In production, it's likely preferable for the menu to display
       * a blank list when the backend API is down, rather than
       * displaying mock data that does not reflect the database state.
       */
      .catch(function(error) {
        return fallbackManuscripts;
      })
      .then(data => {
        let ms = data.map(manuscript => {
          return <option key={manuscript.id}>{manuscript.shelfmark}</option>;
        });
        this.setState({ manuscripts: ms });
      });
  }
  render() {
    return <select multiple>{this.state.manuscripts}</select>;
  }
}

export default ManuscriptsLoader;
