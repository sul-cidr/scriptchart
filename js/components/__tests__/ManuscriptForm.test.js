/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ManuscriptForm from '../ManuscriptForm';
import { defaultManuscripts } from '../ManuscriptsLoader';


describe('ManuscriptForm test', () => {
  it('ManuscriptForm should match snapshot', () => {
    const component = renderer.create(<ManuscriptForm manuscripts={defaultManuscripts}
       />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
