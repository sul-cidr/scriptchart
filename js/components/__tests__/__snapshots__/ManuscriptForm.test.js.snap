// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManuscriptForm test ManuscriptForm should match snapshot 1`] = `
<form
  className="manuscript-form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="control"
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <label
        className="control"
        htmlFor="selectedShelfmarks"
      >
        Select manuscripts:
         
      </label>
      <span
        className="button is-small"
        onClick={[Function]}
      >
        All/None
      </span>
    </div>
    <div
      className="select is-multiple"
    >
      <select
        id="selectedShelfmarks"
        multiple={true}
        name="selectedShelfmarks"
        onChange={[Function]}
        size="8"
        type="string"
        value={Array []}
      >
        <option
          value="Vat. Syr. 092"
        >
          Vat. Syr. 092
           (
          823
          )
        </option>
        <option
          value="Vat. Syr. 112"
        >
          Vat. Syr. 112
           (
          552
          )
        </option>
        <option
          value="Vat. Syr. 140"
        >
          Vat. Syr. 140
           (
          528
          )
        </option>
        <option
          value="BL. Add. 17224"
        >
          BL. Add. 17224
           (
          1173
          )
        </option>
        <option
          value="BL. Add. 12144"
        >
          BL. Add. 12144
           (
          1081
          )
        </option>
        <option
          value="BL. Add. 12134"
        >
          BL. Add. 12134
           (
          697
          )
        </option>
        <option
          value="BL. Add. 14445"
        >
          BL. Add. 14445
           (
          532
          )
        </option>
        <option
          value="BL. Add. 12146"
        >
          BL. Add. 12146
           (
          1007
          )
        </option>
        <option
          value="BL. Or. 8732"
        >
          BL. Or. 8732
           (
          770
          )
        </option>
        <option
          value="BL. Or. 8606"
        >
          BL. Or. 8606
           (
          723
          )
        </option>
        <option
          value="BL. Add. 17102"
        >
          BL. Add. 17102
           (
          598-599
          )
        </option>
        <option
          value="BL. Add. 17107"
        >
          BL. Add. 17107
           (
          540-541
          )
        </option>
        <option
          value="BL. Add. 12175"
        >
          BL. Add. 12175
           (
          533-534
          )
        </option>
      </select>
    </div>
    <div
      className="control"
    >
      <p>
        Order manuscripts by
      </p>
      <label
        className="radio"
        htmlFor="shelfmarkSort"
      >
        <input
          checked={true}
          id="shelfmarkSort"
          onChange={[Function]}
          type="radio"
          value="shelfmark"
        />
         
        Shelfmark |
      </label>
      <label
        className="radio"
        htmlFor="dateSort"
      >
        <input
          checked={false}
          id="dateSort"
          onChange={[Function]}
          type="radio"
          value="date"
        />
         
        Date
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <label
        className="control"
      >
        Select letters: 
      </label>
      <span
        className="button is-small"
        onClick={[Function]}
      >
        All/None
      </span>
    </div>
    <div
      className="buttons are-small"
    >
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Alaph (Angular)"
        >
          ܐ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="Alaph (Round)"
        >
          ܐ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Beth"
        >
          ܒ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Gamal"
        >
          ܓ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Dalath (Angular)"
        >
          ܕ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="Dalath (Round)"
        >
          ܕ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="He (Angular)"
        >
          ܗ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="He (Round)"
        >
          ܗ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Waw"
        >
          ܘ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Zain"
        >
          ܙ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Heth"
        >
          ܚ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Teth"
        >
          ܛ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Yudh (Connected)"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/yudh_connected.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Yudh (Stand-alone)"
        >
          ܝ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Kaph"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/kaph.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Kaph (Final)"
        >
          ܟ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Lamadh (Final, open)"
        >
          ܠ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Lamadh"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/lamadh.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="Lamadh (Final, closed)"
        >
          ܠ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Mim"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/mim.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Mim (Final)"
        >
          ܡ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Nun"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/nun.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Nun (Final, connected)"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/nun_final.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Nun (Final, unconnected)"
        >
          ܢ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Semkath"
        >
          ܣ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Ayin"
        >
          ܥ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Pe"
        >
          ܦ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Sadhe"
        >
          ܨ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Qaph"
        >
          ܩ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Rish (Angular)"
        >
          ܪ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="Rish (Round)"
        >
          ܪ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Shin"
        >
          ܫ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "EstrangeloEdessa",
            }
          }
          title="Taw (Looped)"
        >
          ܬ
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "verticalAlign": "center",
            }
          }
          title="Taw (Triangular)"
        >
          <img
            className="button-svg "
            src="../assets/font_glyphs/taw_triangular.svg"
          />
        </div>
      </span>
      <span
        className="button is-outlined"
        onClick={[Function]}
      >
        <div
          className="button-letter"
          style={
            Object {
              "fontFamily": "SertoJerusalem",
            }
          }
          title="Taw (L-shaped)"
        >
          ܬ
        </div>
      </span>
    </div>
  </div>
  <div
    className="field is-horizontal flex-row"
  >
    <label
      className="control"
      htmlFor="letterExamples"
    >
      Number of examples:
       
    </label>
    <div
      className="select is-small"
      style={
        Object {
          "marginLeft": "5px",
        }
      }
    >
      <select
        id="letterExamples"
        name="letterExamples"
        onChange={[Function]}
        type="number"
        value={3}
      >
        <option>
          1
        </option>
        <option>
          3
        </option>
        <option>
          5
        </option>
      </select>
    </div>
  </div>
  <div
    className="control"
  >
    <p>
      Show letter images:
    </p>
    <label
      className="checkbox"
      htmlFor="showBinarized"
    >
      <input
        checked={true}
        id="showBinarized"
        name="showBinarized"
        onChange={[Function]}
        type="checkbox"
      />
       Trimmed |
    </label>
    <label
      className="checkbox"
      htmlFor="showCropped"
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      <input
        checked={false}
        id="showCropped"
        name="showCropped"
        onChange={[Function]}
        type="checkbox"
      />
       Untrimmed
    </label>
  </div>
  <div
    className="field is-horizontal flex-row"
  >
    <label
      className="control"
      htmlFor="imageSize"
    >
      Image size:
       
    </label>
    <div
      className="select is-small"
      style={
        Object {
          "marginLeft": "5px",
        }
      }
    >
      <select
        id="imageSize"
        name="imageSize"
        onChange={[Function]}
        type="string"
        value="Medium"
      >
        <option>
          Small
        </option>
        <option>
          Medium
        </option>
        <option>
          Large
        </option>
      </select>
    </div>
  </div>
  <div
    className="control"
  >
    Letter in context:
    <ul>
      <li>
        <label
          className="radio"
          htmlFor="hoverContext"
        >
          <input
            checked={true}
            id="hoverContext"
            onChange={[Function]}
            type="radio"
            value="hover"
          />
           
          Show on hover
        </label>
      </li>
      <li>
        <label
          className="radio"
          htmlFor="clickContext"
        >
          <input
            checked={false}
            id="clickContext"
            onChange={[Function]}
            type="radio"
            value="click"
          />
           
          Show on click
        </label>
      </li>
    </ul>
  </div>
  <div
    className="field is-horizontal flex-row"
  >
    <label
      className="control"
      htmlFor="contextSize"
    >
      Context size:
       
    </label>
    <div
      className="select is-small"
      style={
        Object {
          "marginLeft": "5px",
        }
      }
    >
      <select
        id="contextSize"
        name="contextSize"
        onChange={[Function]}
        type="string"
        value="large"
      >
        <option
          value="small"
        >
          Small
        </option>
        <option
          value="med"
        >
          Medium
        </option>
        <option
          value="large"
        >
          Large
        </option>
      </select>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        className="button is-link"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
